### ISSUE TITLE
Should 'between' contain entries that are partially inside the range?

### ISSUE THREAD

**Issue by mwebler** [mifi/ical-expander#6]
Hi there, first of all, thank you, your library have saved us a lot of time 😄 

I have one question about how the between function works... On lines 76 and 96 you have this condition to filter the events in the range:

```
(!before || endTime <= before.getTime()) &&
(!after || startTime >= after.getTime())
``` 

In this case, you are filtering only events that start after `before` AND ends before `after` (starts and ends in the range).
Shouldn't this case return also events that are just partially inside the range? 

i.e. event starts before the range but ends inside it (or starts inside it and ends after the range):
![eventssample](https://imgur.com/download/NjzswhQ "Event examples")

In this case, all 3 events should be included because all of them occur during the requested range (even if they start or end out of the range)
One use case where this would be used is asking for 'What I am doing today?'. If there is an event that starts at 11pm and ends in the next day at 2am, I would want this event to be shown, because it starts on the requested range. This is the event1 case in the example.

What do you think about it? If you agree I can submit a PR for that... 
If it is designed to work like this, no problem as well. Then I will probably just make a fork for my projects 😄 

Thanks!``

**Comment by mifi** [comment]
Hey!

Nice idea! I originally just added -24h to +24h on the between dates before calling and then I did the filtering afterwards, which fit my use case. But I agree that this is a better way of doing it.

If can implement this in a PR and see if the existing tests still work, and add a new test case that tests this behaviour, it would be nice. And maybe put your nice graph in readme with your explanation :) (u can use markdown blockquote for monospace)

I hope people are not relying too much on the old behaviour. We can make it a major version bump (breaking change.) and keep the signatures of the methods the same.

**Comment by mifi** [comment]
implemented in v2.0.0
